core/src/main/java/org/apache/accumulo/core/fate/ZooStore.java

```diff
@@ -202,12 +202,10 @@ public long reserve() {
             if (deferred.isEmpty()) {
               this.wait(5000);
             } else {
-              long currTime = System.nanoTime();
-              long minWait =
-                  deferred.values().stream().mapToLong(l -> l - currTime).min().getAsLong();
-              long waitTime = TimeUnit.MILLISECONDS.convert(minWait, TimeUnit.NANOSECONDS);
-              if (waitTime > 0) {
-                this.wait(Math.min(waitTime, 5000));
+              final long now = System.nanoTime();
+              long minWait = deferred.values().stream().mapToLong(l -> l - now).min().orElseThrow();
+              if (minWait > 0) {
```

Comment:
This code was previously converting nanos to millis.  Seems like that is still needed

Reply:
Good catch. Fixed in 55e1c1e.